test_launch_ros/test/test_launch_ros/actions/test_set_parameter.py

```diff
@@ -42,32 +45,27 @@ def get_set_parameter_test_parameters():
     return [
         pytest.param(
             [('my_param', '2')],  # to set
-            {'my_param': '2'},  # expected
+            [('my_param', '2')],  # expected
             id='One param'
         ),
-        pytest.param(
-            [('my_param', '2'), ('another_param', '2'), ('my_param', '3')],
-            {'my_param': '3', 'another_param': '2'},
-            id='Two params, overriding one'
-        ),
```

Comment:
Why remove this test case?

Reply:
when ``my_param`` is passed again with a different value, it can be reassigned due to ``ros__params`` being a dictionary, with ``my_param`` as a key, and ``3`` as its value. Since we shifted to a list of tuples to preserve order, this would be passed as ``-p my_param:=2 -p another_param:=2 -p my_param:=3`` which would override ``my_param`` anyway.  Thus this test seemed redundant, and won't work as it was originally intended.

Reply:
I did try adding a logic to replace the original declaration when it is overridden, but it seemed redundant as ``rcl`` itself is doing it anyway down the line

Reply:
I understand, thanks for explaining; makes sense to me, we can leave it as-is.